### PR TITLE
(#649) Enumerate sensu_enterprise_dashboard_config instances

### LINKED ISSUE #649: implement an `instances` method for the sensu_enterprise_dashboard_api_config `json` provider
This would facilitate, e.g, using the `resources` resource to `purge` any unmanaged API endpoints from the config

See https://docs.puppet.com/puppet/4.10/provider_development.html#providerresource-api

### PR DESCRIPTION
**Pull request by jeffmccune**
Without this patch the `resources` resource cannot be used to purge unmanaged
API endpoints as requested in #649.  This patch implements the instances class
method providing this functionality.

resolves #649

### CODE REVIEW

**Comment by ghoneycutt** [comment]
What's with the empty commit?

**Comment by jeffmccune** [comment]
> What's with the empty commit?

Made the empty commit to trigger Waffle so I'd be assigned the issue and it'd move over to in-progress.

**Comment by jeffmccune** [comment]
This is ready for review / merge.

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 require 'puppetlabs_spec_helper/module_spec_helper'
+require 'pry'
```
Comment (ghoneycutt) [inline spec/spec_helper.rb]: This should be removed. It is what's causing Travis to fail.

File: Gemfile [hunk 2]
```diff
@@ -50,4 +50,8 @@ group :development do
   gem 'listen', '~> 3.0.0', :require => false
 end
 
+if File.exists? "#{__FILE__}.local"
```
Comment (ghoneycutt) [inline Gemfile]: This would be more clear if it read `Gemfile.local` instead of being variable, since the name of the Gemfile will not change.

**Comment by ghoneycutt** [comment]
Could you please add documentation of this new feature in the README.

**Comment by ghoneycutt** [comment]
Thanks @jeffmccune 

Could you add the README portion and then this is good to go.

**Comment by jeffmccune** [comment]
@ghoneycutt README updated, review comments have been addressed.  Should be ready for merge / release now.

**Comment by ghoneycutt** [comment]
Released in v2.16.0

**Pull request merged**